nemo_rl/data/multimodal_utils.py

```diff
@@ -5,120 +5,60 @@
-        assert len(set(dim)) == 1, "All packed multimodal data must have the same dim_to_pack"
-        dim = dim[0]
-        tensors = [item.tensor for item in packed_data]
-        # packed batch
-        batch = cls.BatchClass(tensors, dim)
-        if return_as_item:
-            return batch.as_packed_data_item()
-        return batch
+class PackedMultimodalData:
+    """Wrapper around a list of torch tensors that contains multimodal data.
 
-class PackedGenericDataBatch:
+    This class is used to wrap a list of tensors containing multimodal data along with a `dim_to_pack` parameter.
+    Since multimodal data can be packed along different dimensions, `dim_to_pack` is used to specify the dimension along which to pack the multimodal data tensors.
+    The multimodal data can be returned as a single packed tensor by calling `as_tensor` which will concatenate the tensors along the `dim_to_pack` dimension.
+    """
 
-    ItemClass = PackedGenericDataItem
-    """Wrapper around a torch tensor that contains multimodal data"""
+    def __init__(self, tensors: Union[torch.Tensor, list[torch.Tensor]], dim_to_pack: int = None) -> None:
+        assert tensors is not None, "Input tensors to PackedMultimodalData cannot be None"
 
-    def __init__(self, tensors: Union[list[torch.Tensor], list[PackedGenericDataItem]], dim_to_pack: Optional[int] = None) -> None:
-        tensorlist: list[torch.Tensor] = []
-        item_dim_to_pack = []
-        for item in tensors:
-            if isinstance(item, torch.Tensor):
-                tensorlist.append(item)
-            elif isinstance(item, PackedGenericDataItem):
-                item_dim_to_pack.append(item.dim_to_pack)
-                tensorlist.append(item.tensor)
-            else:
-                raise ValueError("tensors must be a list of torch.Tensors or a list of PackedGenericDataItem objects")
-        self.tensors: list[torch.Tensor] = tensorlist
-
-        if len(set(item_dim_to_pack)) > 1:
-            raise ValueError("All items in the batch must have the same dim_to_pack")
-        if len(set(item_dim_to_pack)) == 1:
-            assert dim_to_pack is None or item_dim_to_pack[0] == dim_to_pack, "dim_to_pack must not provided, or must be equal as each item in the batch"
-            dim_to_pack = item_dim_to_pack[0]
+        if isinstance(tensors, torch.Tensor):
+            self.tensors: list[torch.Tensor] = [tensors]
+        elif isinstance(tensors, list):
+            assert len(tensors) > 0, "Input tensors to PackedMultimodalData must be a non-empty list"
+            self.tensors: list[torch.Tensor] = tensors
         else:
-            # empty string, make sure dim_to_pack is provided
-            assert dim_to_pack is not None, "dim_to_pack must be provided for empty batch"
-            
+            raise ValueError(f"Unsupported type for input tensors to PackedMultimodalData: {type(tensors)}")
         self.dim_to_pack = dim_to_pack 
```

Comment:
we probably should not have a default value for `dim_to_pack` anymore right? (especially `None` doesn't make sense anymore since I had that condition in case it could be inferred from the underlying `PackedDataItem`